wagtail/wagtailcore/fields.py

```diff
@@ -12,31 +12,55 @@
 from wagtail.wagtailcore.blocks import Block, StreamBlock, StreamValue, BlockField
 
 
-class RichTextArea(WidgetWithScript, forms.Textarea):
+class BaseTextAreaWidget(WidgetWithScript, forms.Textarea):
+
+    def __init__(self, editor_config=None, **kwargs):
+        # set the base editor configuration if existent
+        self.editor_config = editor_config
+        super(BaseTextAreaWidget, self).__init__(**kwargs)
+
     def get_panel(self):
-        from wagtail.wagtailadmin.edit_handlers import RichTextFieldPanel
-        return RichTextFieldPanel
+        raise NotImplementedError(
+            "Class %s doesn't implement a required Method()" % (self.__class__.__name__))
```

Comment:
Would be great to say what method developer need to implement. I mean message may look like this:
```python
"Class %s doesn't implement a required get_panel method" % (self.__class__.__name__))
```